### PR TITLE
fix(tests): Fix broken deepEqualsWithBigNumber implementation

### PR DESCRIPTION
**Pull request by nicholaspai**
`deepEqualsWithBigNumber` doesn't actually deeply compare objects as expected because any Object.toString() === Object.toString().

The easiest way I see to fix this is to replace the implementation using a combination of assert.deepStrictEquals and lodash.omit

I had to fix broken tests as a result

### CODE REVIEW

File: test/ConfigStoreClient.ts [hunk 1]
```diff
@@ -64,8 +71,13 @@ describe("AcrossConfigStoreClient", async function () {
     configStore = await (await getContractFactory("AcrossConfigStore", owner)).deploy();
     const receipt = await configStore.deployTransaction.wait();
     const eventSearchConfig = { fromBlock: receipt.blockNumber };
-    configStoreClient = new MockConfigStoreClient(createSpyLogger().spyLogger, configStore, eventSearchConfig);
-    configStoreClient.setConfigStoreVersion(0);
+    configStoreClient = new ConfigStoreClient(
```
Comment (nicholaspai) [inline test/ConfigStoreClient.ts:74]: Unrelated to the `deepEquals` bug, I can't get any of the `MockConfigStoreClient` tests to work. It only works when I use the non mocked version. @pxrl i recall you worked on this recently, any ideas if I'm using it incorrectly?

Comment (pxrl) [reply test/ConfigStoreClient.ts:74]: I _think_ this might be related to running tests in parallel - is that how you're testing @nicholaspai ?

Comment (pxrl) [reply test/ConfigStoreClient.ts]: Confirmed unrelated to running the tests in parallel - there seems to have been a local issue with `yarn link`.

File: test/Dataworker.loadData.ts
[the remark this answers is not in the document]
Comment (nicholaspai) [reply test/Dataworker.loadData.ts:1]: theyve always been broken, yep

**Pull request merged**